src/yunohost/user.py

```diff
@@ -566,6 +607,225 @@ def user_info(username):
+        width = 20
+        bar = int(progress.nb * width / total)
+        bar = "[" + "#" * bar + "." * (width - bar) + "]"
+        if info:
+            bar += " > " + info
+        if progress.old == bar:
+            return
+        progress.old = bar
+        logger.info(bar)
+    progress.nb = 0
+    progress.old = ""
+
+    def on_failure(user, exception):
+        result['errors'] += 1
+        logger.error(user + ': ' + str(exception))
+
+    def update(user, info=False):
+        remove_alias = None
+        remove_forward = None
+        if info:
+            user['mail'] = None if info['mail'] == user['mail'] else user['mail']
+            remove_alias = list(set(info['mail-alias']) - set(user['mail-alias']))
+            remove_forward = list(set(info['mail-forward']) - set(user['mail-forward']))
+            user['mail-alias'] = list(set(user['mail-alias']) - set(info['mail-alias']))
+            user['mail-forward'] = list(set(user['mail-forward']) - set(info['mail-forward']))
+            for group, infos in user_group_list()["groups"].items():
+                if group == "all_users":
+                    continue
+                # If the user is in this group (and it's not the primary group),
+                # remove the member from the group
+                if user['username'] != group and user['username'] in infos["members"]:
+                    user_group_update(group, remove=user['username'], sync_perm=False, imported=True)
+
+        user_update(user['username'],
+                user['firstname'], user['lastname'],
+                user['mail'], user['password'],
+                mailbox_quota=user['mailbox-quota'],
+                mail=user['mail'], add_mailalias=user['mail-alias'],
+                remove_mailalias=remove_alias,
+                remove_mailforward=remove_forward,
+                add_mailforward=user['mail-forward'], imported=True)
+
+        for group in user['groups']:
+            user_group_update(group, add=user['username'], sync_perm=False, imported=True)
+
+    users = user_list(CSV_FIELDNAMES)['users']
+    operation_logger.start()
+    # We do delete and update before to avoid mail uniqueness issues
+    for user in actions['deleted']:
+        try:
+            user_delete(user, purge=True, imported=True)
+            result['deleted'] += 1
+        except YunohostError as e:
+            on_failure(user, e)
+        progress("Deletion")
+
+    for user in actions['updated']:
+        try:
+            update(user, users[user['username']])
+            result['updated'] += 1
+        except YunohostError as e:
+            on_failure(user['username'], e)
+        progress("Update")
+
+    for user in actions['created']:
+        try:
+            user_create(user['username'],
+                        user['firstname'], user['lastname'],
+                        user['domain'], user['password'],
+                        user['mailbox-quota'], imported=True)
+            update(user)
```

Comment:
uuuuuuuuuuh i guess that simplifies the code for groups management but this seems super suboptimal in tems of perfs idk ? Gotta think about it